ddtrace/internal/_encoding.pyx

```diff
@@ -217,10 +217,14 @@ cdef class MsgpackEncoderBase(BufferedEncoder):
         ret = msgpack_pack_array(&self.pk, L)
         if ret != 0: raise RuntimeError("Couldn't pack trace")
 
-        if L > 0 and trace[0].context is not None:
-            dd_origin = trace[0].context.dd_origin
-        else:
-            dd_origin = None
+        if L > 0 and trace[0].context is not None and trace[0].context.dd_origin is not None:
+            IF PY_MAJOR_VERSION >= 3:
+                # To cast as a char array, the unicode string needs to be converted to a bytestring
+                # with the reference to the Python bytestring as long as the char array is in use.
+                dd_origin_bytestring = trace[0].context.dd_origin.encode("UTF-8")
+                dd_origin = dd_origin_bytestring
+            ELSE:
+                dd_origin = trace[0].context.dd_origin
```

Comment:
we do this elsewhere, might be working having an inline cdef function to convert PyObject to a string?

Reply:
Just checked the other usage in `pack_text()`, that one directly encodes the text into the msgpack buffer and is handling the cases of py2 unicode vs py3 unicode, so I don't think it's applicable here